Stack/Opc.Ua.Core/Stack/Bindings/BufferManager.cs

```diff
@@ -121,10 +121,13 @@ public class BufferManager
         /// <param name="maxBufferSize">Max size of the buffer.</param>
         public BufferManager(string name, int maxBufferSize)
         {
-            int maxArrayLength = maxBufferSize + m_cookieLength;
+            m_arrayPool = maxBufferSize <= 1024 * 1024
+                ? ArrayPool<byte>.Shared
+                : ArrayPool<byte>.Create(maxBufferSize, 32);
+#if TRACK_MEMORY
```

Comment:
use maxArrayLength, bring cookie back etc. Its needed for runtime check of buffer usage.